src/MongoDB.Driver.Core/Core/Compression/CompressorTypeMapping.cs

```diff
@@ -0,0 +1,51 @@
+﻿/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+namespace MongoDB.Driver.Core.Compression
+{
+    internal static class CompressorTypeMapping
+    {
+        public static string ToServerName(CompressorType compressorType)
+        {
+            switch (compressorType)
+            {
+                case CompressorType.Noop:
+                    return "noop";
+                case CompressorType.Zlib:
+                    return "zlib";
+                case CompressorType.Snappy:
+                    return "snappy";
+                case CompressorType.ZStandard:
+                    return "zstd";
+                default:
+                    return compressorType.ToString().ToLowerInvariant();
```

Comment:
I think this is wrong. The server supports receiving unsupported compression keys. In this case, his response won't contain any compressions and as a result `AvailableCompression` list will be empty.
It can be also a minor BC since previously we didn't throw any exception for arguments like this:

                new MongoClientSettings()
                {
                    Compressors = new[] { new CompressorConfiguration((Core.Compression.CompressorType)5)}
                }

Reply:
Done, @jyemin can you take a quick look at this question. 
it seemed to me that we should not throw any exceptions in the handshake step in case if we use unexpected compressorType regardless how this value was received (even if it was with this weird code as the above). NOTE: we cannot have such a situation other than with weird code like `(Core.Compression.CompressorType)5`

Reply:
The ONLY way to have an INVALID value for the CompressorType enum is by casting an integer as in your example above.

If we don't throw an exception we will instead be sending the string "5" to the server, which makes no sense.

I would say that the language that says that unknown compressor types should be ignored applies to the connection string (where the compressor types are represented using strings that could have ANY value) and not to configuration in C# code where we have an enum with a FINITE set of valid values.